### PR TITLE
feat(sdk): enable component compilation via Compiler

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Enables compilation of component IR via the `Compiler.compile` method.

Note: 7c6fb85dab6a244165715e0b1b7245809c21fee9 (and associated d595a49e6cd4117fbc4f30404a3b4b562f94d96c) is, while not a breaking change, a very significant change and deserves careful review and discussion.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more 
about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/compiler/compiler.py [hunk 1]
```diff
@@ -42,16 +42,48 @@
 
 _GroupOrTask = Union[tasks_group.TasksGroup, pipeline_task.PipelineTask]
 
+CompileSignature = Callable[..., None]
+
+
+def _alias_deprecated_params(
```
Comment (ji-yaqi) [inline sdk/python/kfp/compiler/compiler.py:48]: Nice util! Wondering if it's beneficial to put in utils folder, so other parts could also use this.

**Review by ji-yaqi — COMMENTED** [review]
LGTM except for a nit, feel free to merge after GCPC parts get resolved.

/approve

File: sdk/python/kfp/compiler/compiler.py (hunk 1 above)
Comment (connor-mccarthy) [reply sdk/python/kfp/compiler/compiler.py:48]: Good idea -- will do!

**Comment by ji-yaqi** [comment]
/lgtm

Thanks!

**Pull request closed without merging**